packages/proof-service/src/domain/create-proof.ts

```diff
@@ -69,15 +77,34 @@ export async function createProof() {
       new Date(task.createdAt.getTime() - 1)
     );
 
+    class ZkDbApp extends getZkDbSmartContract(task.database, merkleHeight) {}
+    const zkDbApp = new ZkDbApp(appPublicKey);
+
+    // TODO: Check if app exists
+
+    const onChainRootState = zkDbApp.state.get();
+    const onChainActionState = zkDbApp.actionState.get();
+
+    assert(onChainRootState.equals(merkleRoot));
+
+    const proofState = new ProofState({
+      actionState: onChainActionState,
+      rootState: onChainRootState,
+    });
+
+    const action = (
+      await zkDbApp.reducer.fetchActions({
+        fromActionState: onChainActionState,
+      })
+    )[0][0];
```

Comment:
This hard code look so ugly `[0][0]` really?

Reply:
We can send multiple actions per transaction, therefore it forms 2-d array. As we fetch actions from a certain state, we take the only first element and it is [0][0]

Reply:
I think, we're better to check for data validity first.

```ts
export function isNotEmptyArray(data: any): bool {
  return Array.isArray(data) && data.length > 0;
}
```

```ts
const allActions =  await zkDbApp.reducer.fetchActions({ fromActionState: onChainActionState });

if( isNotEmptyArray(allActions) && isNotEmptyArray(allActions[0]) ) {
  const [[action]] = allActions;
} 
throw new Error('Unformatted action data');
```

Reply:
Done